### DISCUSSION TITLE
Question about making modifications to wind shear

### DISCUSSION THREAD

**Discussion by michaelxu3** [contrailcirrus/pycontrails#144] [Q&A]
Hi, I've been using CoCiP to help with an experiment where I look at systematic biases in the ERA5 wind data and their effects on the outputs of contrail models. Is there a fast "switch" to allow me to get CoCiP to ignore the ERA5 input and set the wind shear (or other meteorological parameters) to a constant? 

If not, I can make modifications to the code locally and am guessing that I just need to overwrite `ds_dz` and related variables in `cocip.py` every timestep and make sure I set the wind shear exponent as well, but just wanted to ask for guidance to make sure I don't overlook anything. I also wanted to ask what the "wind shear normal" or `dsn_dz` (in the cocip model) specifically is and what it's used for.

**Comment by zebengberg** [comment]
Hi @michaelxu3 

There's no easy switch to override the meteorology. I think the quickest way to achieve what your asking would be to create a custom `MetDataset` (a wrapper around [`xr.Dataset`](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.html)) in which you set certain wind fields to constants. If you were to set `eastward_wind` and `northward_wind` to constants, you may be able to simulate the model without wind shear. (Cocip requires [these meteorology fields](https://py.contrails.org/api/pycontrails.models.cocip.Cocip.html#pycontrails.models.cocip.Cocip).)

The [`dsn_dz`](https://py.contrails.org/api/pycontrails.models.cocip.wind_shear.html#pycontrails.models.cocip.wind_shear.wind_shear_normal) variable is derivative in the z direction of the wind normal to the contrail segment. So this tells you how the contrail is shearing normal to its direction.

You may also be able to monkey patch the [wind shear function](https://github.com/contrailcirrus/pycontrails/blob/main/pycontrails/models/cocip/wind_shear.py#L49-L86) to disable wind shear effects. I haven't tried it myself, but something like this could work.

```python
def override_wind_shear_normal(u_wind_top, u_wind_btm, v_wind_top, v_wind_btm, cos_a, sin_a, dz):
    return np.zeros_like(u_wind_top)

from pycontrails.models.cocip import wind_shear
wind_shear.wind_shear_normal = override_wind_shear_normal
...
```
